Registry migration page (#781)
wandb/docs: docs/guides/registry/model_registry_eol.md

---
displayed_sidebar: default
---

> # Model Registry end of life

Reviewer: we s
```suggestion
# Legacy Model Registry end of life
```

> W&B will migrate assets from [W&B Model Registry](../model_registry/intro.md) to [W&B Registry](./intro.md) in mid-October 2024.By the end of 2024 you will no longer be able to access W&B Model Registry. An exact date will be shared at least 3 weeks in advance to all W&B Model Registry users. 

Reviewer: ```suggestion
W&B will migrate assets from the legacy [W&B Model Registry](../model_registry/intro.md) to [W&B Registry](./intro.md) in mid-October 2024. By the end of 2024 you will no longer be able to access W&B Model Registry. An exact date will be shared at least 3 weeks in advance to all W&B Model Registry users. 
```

This guide will cover what changes will occur, how W&B will migrate assets, along with information on how to prepare your current assets for the migration process.

Contact support@wandb.com for more information about the migration process.

## What is changing?

The following tables highlights the major changes that will happen from the legacy Model Registry to the W&B Registry:

|               | legacy W&B Model Registry | W&B Registry |
| -----         | ----- | ----- |
| Artifact visibility| Team level | Organization level |
| Name change | A set of pointers (links) to model versions are called *registered models*. | A set of pointers (links) to artifact versions are called *collections*. | 

> | `wandb.init.link_model` | Model Registry specific API | No longer supported |

Reviewer: ```suggestion
| `wandb.init.link_model` | Model Registry specific API | currently only compatible with legacy model registry |
```

<!-- Registered models in the legacy W&B Model Registry are renamed to Collections in W&B Registry. -->

<!-- Artifacts in the new W&B Registry have organization level scope. Artifacts in the legacy W&B Model Registry have team level scope.  -->

## Preparing for the migration

> ### What is migrating?

Reviewer: can we add the diagram into this section? I've shared it with a few customers and they've noted it's a very helpful visual. 
![model-reg-migration-new](https://github.com/user-attachments/assets/fdf70509-08d4-4487-8178-6312446908c4)

> W&B will migrate [registered models](../model_registry/create-registered-model.md) and the [artifact versions](../model_registry/link-model-version.md) that are in your current W&B Model Registry, to W&B Registry as collections. More specifically, your registered models will be stored within within a registry called **Model**.

Reviewer: ```suggestion
W&B will migrate [registered models](../model_registry/create-registered-model.md) (now called collections) and the [artifact versions](../model_registry/link-model-version.md) that are in your current W&B Model Registry, to W&B Registry as collections. More specifically, your registered models (now called collections) will be stored within within a registry called **Model**.
```

![](/images/registry/mode_reg_eol.png)

> :::info Team visibility to organization visibility

Author: Let's make this super visible and make it's own section about team -> org visibility?

Reviewer: that sounds good!

:::

### Artifact path changes

After the migration the paths to artifacts linked within collections will change from:

```python
team-name/model-registry/collection-name:v0
```
to:

```python
org-name/wandb-registry-model/team-name_collection-name:v0
```

:::info Action required
Update any programmatic references to legacy Model Registry paths within your code to avoid disruptions. 
:::

For example, suppose that your current code looks like:

```python
import wandb 

team_entity = "team_awesome"
registered_model_name = "mnist"
version = "latest"

# Initialize a run
run = wandb.init(entity=team_entity)

# The path where you want to link the artifact version to
artifact_version = f"{team_entity}/model-registry/{registered_model_name}:{version}"

# Link an artifact version to a registered model called mnist within the Model Registry
run.link_artifact(artifact_or_name=artifact_version)
```

In preparation of the migration, you will need to ensure that you specify the entity of the organization and provide a name for the collection (formally known as registered model in legacy W&B Model Registry):

```python
import wandb

> org_entity = "ReviewCo" # Use organization entity

Reviewer: ```suggestion
org_entity = "reviewco" # Use organization entity
```

| Method | Parameter | 
| -- | -- | 
| [`wandb.run.link_artifact()`](../../ref/python/run.md#link_artifact) | `target_path` | 
| [`wandb.Artifact.link()`](../../ref/python/artifact.md#link) |`target_path` | 
| [`wandb.Api().run.use_artifact()`](../../ref/python/public-api/run.md#use_artifact)| `artifact`| 
| [`wandb.Api().artifact()`](../../ref/python/public-api/api.md#artifact) | `name` | 

### Registered model attributes

W&B will migrate the following attributes in your current registered models to collections:

- Descriptions
- Tags
- Aliases

### Automations

<!-- Noah to update automation links -->

Any automations that your team created will be migrated to the new registry. The automations will use the same registered model/collection that you created the automation for. 

### Deprecated features 

* [Protected aliases](../model_registry/access_controls.md#add-protected-aliases): Aliases will no longer have a protected status available. Instead, you can go to a registry’s settings to configure which users have viewer, member, or admin status. All aliases that were previously protected will become standard aliases. 

> <!-- *  Service accounts: Team Service accounts will temporarily not be able to interact programmatically with Registry. This feature will be re-introduced before the migration, however.  -->

Reviewer: I see this was commented out but we definitely should keep it in here!

Author: Oops. My bad

Author: Can we add some tentative dates? e.g. Team Service accounts are temporarily unable to interact programmatically with Registry? W&B will enable programmatic access to service account in [INSERT tent. date].

Reviewer: I don't have tentative dates and don't want to commit on any since this is a dependency on the platform team. We're exploring a tentative workaround - see my updated below in the later commit

## During the migration

Once the migration begins, there will be a write-block on the legacy Model Registry. This write-block means that users will not be able to do the following in the legacy Model Registry:

- Create  and delete collections (formerly known as registered models)
- Link new versions and unlink existing versions
- Add and delete aliases to artifacts in the registry
- Add and delete tags to collections in the registry
- Add and delete automations

The migration will last for approximately 24-48 hours and will be conducted during non-business hours to minimize disruption. The legacy Model Registry will remain in a read-only state after the migration.

## After the migration

> After the migration all artifact versions, collections, descriptions, automations, and tags will be migrated to W&B Registry. However, action history will not be transferred.

Reviewer: ```suggestion
After the migration all artifact versions, collections, descriptions, automations, and tags will be migrated to W&B Registry. Action history will be preserved, including a new action that is added to represent that a linked version was migrated, with the ability to navigate to the action history prior to the migration. 
```

Author: A bit of a run on sentence, broke it down into three: "Action history will be preserved. You will still be able to navigate to the action history prior to the migration. In addition, W&B will add a new action to represent which linked version are migrated."

:::

For example, suppose that your current code looks like:

```python
import wandb 

team_entity = "team_awesome"
registered_model_name = "mnist"
version = "latest"

# Initialize a run
run = wandb.init(entity=team_entity)

# The path where you want to link the artifact version to
artifact_version = f"{team_entity}/model-registry/{registered_model_name}:{version}"

# Link an artifact version to a registered model called mnist within the Model Registry
run.link_artifact(artifact_or_name=artifact_version)
```

In preparation of the migration, you will need to ensure that you specify the entity of the organization and provide a name for the collection (formally known as registered model in legacy W&B Model Registry):

```python
import wandb

org_entity = "reviewco" # Use organization entity
collection = "Trained_animal_models" # Registered models now called collections
version = "latest"

run = wandb.init(entity=org_entity)

artifact_version = f"{org_entity}/wandb-registry-model/{collection}:{version}"

# Link an artifact to a collection called Trained_animals_models within the new Models registry

> run.link_artifact(artifact_or_name = artifact_version)

Reviewer: @ngrayluna we are missing the target path here - link artifact takes in the artifact you want to to link and the path you want to link it to

Reviewer: https://docs.wandb.ai/ref/python/run#link_artifact

```

The proceeding table lists the parameters that you need to update the path name for:

| Method | Parameter | 
| -- | -- | 
| [`wandb.run.link_artifact()`](../../ref/python/run.md#link_artifact) | `target_path` | 
| [`wandb.Artifact.link()`](../../ref/python/artifact.md#link) |`target_path` | 
| [`wandb.Api().run.use_artifact()`](../../ref/python/public-api/run.md#use_artifact)| `artifact`| 
| [`wandb.Api().artifact()`](../../ref/python/public-api/api.md#artifact) | `name` | 

> ### Registered model attributes

Reviewer: ```suggestion
### Collection attributes
```

W&B will migrate the following attributes in your current registered models to collections:

- Descriptions
- Tags
- Aliases

### Automations

<!-- Noah to update automation links -->

Any automations that your team created will be migrated to the new registry. The automations will use the same registered model/collection that you created the automation for. 

### Deprecated features 

* [Protected aliases](../model_registry/access_controls.md#add-protected-aliases): Aliases will no longer have a protected status available. Instead, you can go to a registry’s settings to configure which users have viewer, member, or admin status. All aliases that were previously protected will become standard aliases. 

> *  Service accounts: Team Service accounts will temporarily not be able to interact programmatically with Registry. This feature will be re-introduced before the migration, however. 

Reviewer: ```suggestion
*  Service accounts: Existing team service accounts will cannot interact programmatically with Registry as the Registry is at the org level. As a result, Organization Service accounts will be made available (this feature is currently in progress) prior to the migration taking place. Please reach out to support@wandb.com to discuss temporary solutions that we recommend. 
```

Author: "Before the migration, as the Registry is at the org level." what does this mean?

Reviewer: oh rip sorry I didn't clean that up. Updated

## During the migration

Once the migration begins, there will be a write-block on the legacy Model Registry. This write-block means that users will not be able to do the following in the legacy Model Registry:

- Create  and delete collections (formerly known as registered models)
- Link new versions and unlink existing versions
- Add and delete aliases to artifacts in the registry
- Add and delete tags to collections in the registry
- Add and delete automations

The migration will last for approximately 24-48 hours and will be conducted during non-business hours to minimize disruption. The legacy Model Registry will remain in a read-only state after the migration.

## After the migration

After the migration all artifact versions, collections, descriptions, automations, and tags will be migrated to W&B Registry. Action history will be preserved. You will still be able to navigate to the action history prior to the migration. In addition, W&B will add a new action to represent which linked version are migrated.

### Using the new Registry

> All organizations should plan to use the new Models registry inside the new W&B Registry. The same capabilities will be supported, including automations, lineage, and aliasing, with new features introduced. 

Reviewer: ```suggestion
All organizations should plan to use the new model registry inside the new W&B Registry. The same capabilities will be supported, including automations, lineage, and aliasing, with new features introduced. Please review [the list](###deprecated-features) above for features that are not supported currently. 
```

<!-- Note the exception [above](https://www.notion.so/Registries-Release-Communications-6270b35a8c244308acfdfadefbc46c0c?pvs=21). -->

By default, organization administrators will be admins of the new model registry and all other orgs users will be viewers to the model registry. To modify any user’s access to the registry - learn more [here](https://docs.wandb.ai/guides/registry/configure_registry#configure-user-roles-in-a-registry).

### Legacy Model Registry
After the migration, the legacy Model Registry UI will be read-only. It will eventually be fully deprecated from the W&B App UI. You will still have view-only access to all linked models in your legacy registry using the following paths:

1. See linked models in this view-only W&B Project:
```python

> [WANDB_BASE_URL]/team_name/model-registry/artifacts/

Reviewer: ```suggestion
[WANDB_BASE_URL]/team_entity_name/model-registry/artifacts/
```

```
2. Programmatically reference a linked model with the following path:
```python

> team_name/model-registry/collection:alias

Reviewer: ```suggestion
team_entity_name/model-registry/collection:alias
```
